src/janus-api/models/feed.js

```diff
@@ -1,455 +1,471 @@
-          start: function() {
-            updateLocalSpeaking(true);
-          },
-          stop: function() {
-            updateLocalSpeaking(false);
-          }
-        });
-      }
-      speakObserver.start();
-      stream = val;
-    };
-
-    /**
-     * Gets janus stream for the feed
-     * @return {object} janus stream
-     */
-    that.getStream = function() { return stream; };
-
-
-    /**
-     * Sets speaking flag
-     * @param {boolean} val - true if feed speaking
-     */
-    that.setSpeaking = function(val) {
-      speaking = val;
-    };
-
-    /**
-     * Gets feed speaking flag
-     * @return {boolean} true if feed speaking
-     */
-    that.getSpeaking = function() { return speaking; };
-
-    /**
-     * Sets if audio is enabled for that feed. Works only for remote ones.
-     */
-    that.setAudioEnabled = function(val) {
-      audioRemoteEnabled = val;
-    };
-
-    /**
-     * Gets if audio is enabled for that feed
-     */
-    that.getAudioEnabled = function() {
-      return that.isEnabled("audio");
-    };
-
-    /**
-     * Sets if video is enabled for that feed. Works only for remote ones.
-     */
-    that.setVideoEnabled = function(val) {
-      videoRemoteEnabled = val;
-    };
-
-    /**
-     * Gets if video is enabled for that feed
-     */
-    that.getVideoEnabled = function() {
-      return that.isEnabled("video");
-    };
-
-    /**
-     * Checks if audio is being currently detected in the local feed
-     *
-     * @returns {Boolean}
-     */
-    that.isVoiceDetected = function() {
-      return speakObserver && speakObserver.isSpeaking();
-    };
-
-    /**
-     * Checks if data channel is open
-     * @returns {boolean}
-     */
-    that.isDataOpen = function() {
-      if (that.connection) {
-        return that.connection.isDataOpen;
+  var picture = null;
+  var speaking = false;
+  var silentSince = Date.now();
+  var videoRemoteEnabled = true;
+  var audioRemoteEnabled = true;
+  var stream = null;
+  var speakObserver = null; // TODO
+
+  function capitalize(str) {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  }
+
+  /**
+   * Checks if a given channel is enabled
+   *
+   * @param {string} channel - "audio" or "video"
+   * @returns {boolean}
+   */
+  that.isEnabled = function(channel) {
+    if (that.isPublisher) {
+      if (that.connection && that.connection.getConfig()) {
+        return that.connection.getConfig()[channel];
       } else {
-        return false;
-      }
-    };
-
-    /**
-     * Checks if the feed is connected to janus.
-     *
-     * @returns {boolean}
-     */
-    that.isConnected = function() {
-      return (that.connection !== null);
-    };
-
-    /**
-     * Disconnects from janus
-     */
-    that.disconnect = function() {
-      if (that.connection) {
-        that.connection.destroy();
-      }
-      if (speakObserver) {
-        speakObserver.destroy();
+        return null;
```

Comment:
Documentation says that a boolean is returned. `return false`?

Reply:
Good point. I prefer to keep the original behavior, so I am updating the docs.